src/ui/osx/TogglDesktop/Feature/TimeEntryEditor/ResizablePopover.swift

```diff
@@ -0,0 +1,213 @@
+        static let CornerHit: CGFloat = 10.0
+        static let SideHit: CGFloat = 4.0
+    }
+
+    private enum Region {
+        case None
+        case Left
+        case Right
+    }
+
+    // MARK: Variables
+
+    private var isResizable: Bool
+    private var min: NSSize
+    private var max: NSSize
+    private var size: NSSize?
+    private var region: Region = .None
+    private var down: NSPoint?
+    private var bottomHeight = CGFloat(20)
+    private var trackLeft: NSView.TrackingRectTag?
+    private var trackRight: NSView.TrackingRectTag?
+    override var contentViewController: NSViewController? {
+        didSet {
+            prepareTracker()
+        }
+    }
+
+    // MARK: Init
+
+    override init() {
+        self.isResizable = false
+        self.min = CGSize.zero
+        self.max = CGSize.zero
+        super.init()
+    }
+
+    init(min: CGSize, max: CGSize) {
+        self.isResizable = true
+        self.min = min
+        self.max = max
+
+        // If not defined, use the screen height
+        if let screen = NSScreen.main {
+            if max.height == 0 {
+                self.max.height = CGFloat(screen.frame.height)
+            }
+            if max.width == 0 {
+                self.max.width = CGFloat(screen.frame.width)
+            }
+        }
+        super.init()
+    }
+
+    required public init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    deinit {
+        clearTrackers()
+    }
+
+    // MARK: Override
+
+    override public func mouseEntered(with event: NSEvent) {
+        guard isResizable else {
+            super.mouseEntered(with: event)
+            return
+        }
+
+        if region == .None {
+            if event.trackingNumber == trackLeft {
+                region = .Left
+            } else if event.trackingNumber == trackRight {
+                region = .Right
+            } else {
+                region = .None
+            }
+
+            setCursor()
+        }
+    }
+
+    override public func mouseExited(with event: NSEvent) {
+        guard isResizable else {
+            super.mouseExited(with: event)
+            return
+        }
+
+        if down == nil {
+            region = .None
+            setCursor()
+        }
+    }
+
+    override public func mouseDown(with event: NSEvent) {
+        guard isResizable else {
+            super.mouseDown(with: event)
+            return
+        }
+
+        self.size = contentSize
+        self.down = NSEvent.mouseLocation
+    }
+
+    override public func mouseDragged(with event: NSEvent) {
+        guard isResizable else {
+            super.mouseDragged(with: event)
+            return
+        }
+
+        if region == .None {
+            return
+        }
+
+        guard let size = size else { return }
+        guard let down = down else { return }
+
+        let location = NSEvent.mouseLocation
+
+        var movedX = (location.x - down.x) * 2
```

Comment:
@NghiaTranUIT Why do you multiply by 2 here?
It resizes twice as fast as the cursor moves. Regular windows resize together with the cursor.

Reply:
After removing the multiplication it works as expected.

Reply:
It's valid reason @skel35. I removed it in the last commit, please review again 👍